packages/host/app/components/operator-mode/create-file-modal.gts

```diff
@@ -730,8 +725,25 @@ export class ${className} extends ${exportName} {
   });
 }
 
-function camelize(name: string) {
-  return capitalize(camelCase(name));
+function convertToClassName(input: string) {
+  // \p{L}: a letter
+  let invalidLeadingCharactersRemoved = camelCase(
+    input.replace(/^[^\p{L}_$]+/u, ''),
+    { pascalCase: true },
+  );
+
+  let className = invalidLeadingCharactersRemoved.replace(
+    // \p{N}: a number
+    /[^\p{L}\p{N}_$]+/gu,
+    '',
+  );
+
+  // make sure we don't collide with a javascript built-in object
+  if (typeof (globalThis as any)[className] !== 'undefined') {
+    className = `${className}0`;
+  }
+
+  return className;
```

Comment:
have you considered using `lodash.deburr`. i think it might do the thing we want here https://lodash.com/docs/4.17.15#deburr

Reply:
I wanted to preserve the accented characters etc, which that strips out. That’s also why I switched away from Lodash’s camel-casing function, because it does the same. I looked all over for something that would sanitise to produce valid identifiers but I couldn’t find a package 😞